docs/en/query_language/functions/introspection.md

```diff
@@ -0,0 +1,224 @@
+# Introspection Functions
+
+You can use functions described in this chapter to introspect ELF and DWARF for query profiling.
+
+!!! warning "Warning"
+    These functions are slow and may impose security considerations.
+
+For proper operation of introspection functions:
+
+- Install the `clickhouse-common-static-dbg` package.
+- Set the [allow_introspection_functions](../../operations/settings/settings.md#settings-allow_introspection_functions) setting to 1.
+
+    For security reasons introspection functions are disabled by default.
+
+ClickHouse saves profiler reports to the [trace_log](../../operations/system_tables.md#system_tables-trace_log) system table. Make sure the table and profiler are configured properly.
+
+## addressToLine {#addresstoline}
+
+Converts virtual memory address inside ClickHouse server process to the filename and the number of a string in ClickHouse source code, using the debug information installed by the `clickhouse-common-static-dbg` package.
+
+**Syntax**
+
+```sql
+addressToLine(address_of_binary_instruction)
+```
+
+**Parameters**
+
+- `address_of_binary_instruction` ([UInt64](../../data_types/int_uint.md)) — Address of code instruction that have been executed at the moment of sampling.
+
+**Returned value**
+
+- Source code filename and the number of string in this file.
+
+Type: [String](../../data_types/string.md).
+
+**Example**
+
+Enabling introspection functions:
+
+```sql
+SET allow_introspection_functions=1
+```
+
+Selecting the first string from the `trace_log` system table:
+
+```sql
+SELECT 
+    *,
```

Comment:
I'd remove this `*` because it makes harder to find the actual result of function currently being described, while using query profiler in practice requires a separate guide

also we could use only one frame instead of whole stacktrace for the same reason